### PR TITLE
Fix linter errors raised by staticcheck

### PR DESCRIPTION
**Pull request by rdimitrov**
This is a small PR separating a few fixes for linter errors that would otherwise be raised by golangci-lint/staticcheck in #234. 

Signed-off-by: Radoslav Dimitrov <dimitrovr@vmware.com>

### CODE REVIEW

File: local_store.go
Comment (joshuagl) [inline local_store.go:435]: We were ignoring file removal failures and now, with this change, we are not? That seems like a reasonable change, are there likely to be any unexpected consequences of this change?

File: local_store.go
Comment (joshuagl) [inline local_store.go:438]: This seems worth implementing, rather than just removing the comment?

**Review by trishankatdatadog — APPROVED** [review]

**Review by joshuagl — COMMENTED** [review]
Thanks for working on fixing inter failures. I'd like a second set of eyes on the `LocalStore` changes.

File: local_store.go
Comment (trishankatdatadog) [reply local_store.go:435]: Good question. Maybe we should add a test case, if it's missing?

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]
ditto Joshua's comments

File: local_store.go
Comment (rdimitrov) [reply local_store.go:438]: Addressed it 👍

File: local_store.go
Comment (rdimitrov) [reply local_store.go:450]: I tested several use cases and I didn't find a reason to break something. It's good to leave this check though and address it again in case someone raises a valid reason to change it back 👍

**Comment by rdimitrov** [comment]
@trishankatdatadog @joshuagl - Thanks again! I've addressed the feedback 👍

**Comment by trishankatdatadog** [comment]
> @trishankatdatadog @joshuagl - Thanks again! I've addressed the feedback 👍

I'll wait for Joshua's approval

**Comment by rdimitrov** [comment]
@trishankatdatadog @joshuagl - The tests caught some side effect of this that it deletes the "targets" folder in case there are no targets at all left in the repository. Do you think that can be a problem? If so, I can update the implementation to skip deleting folders named "targets".

**Comment by joshuagl** [comment]
> @trishankatdatadog @joshuagl - The tests caught some side effect of this that it deletes the "targets" folder in case there are no targets at all left in the repository. Do you think that can be a problem? If so, I can update the implementation to skip deleting folders named "targets".

@trishankatdatadog @ethan-lowman-dd do you anticipate this behaviour causing problems, based on your experience deploying go-tuf?

**Comment by trishankatdatadog** [comment]
> @trishankatdatadog @ethan-lowman-dd do you anticipate this behaviour causing problems, based on your experience deploying go-tuf?

Is it really more dangerous to delete an empty folder than a target file? BTW, what folder is it: is it specific to that target, or is it for all targets?

**Comment by rdimitrov** [comment]
> > @trishankatdatadog @ethan-lowman-dd do you anticipate this behaviour causing problems, based on your experience deploying go-tuf?
> 
> Is it really more dangerous to delete an empty folder than a target file? BTW, what folder is it: is it specific to that target, or is it for all targets?

Previously, go-tuf ignored deleting folders without any targets in it. Meaning if we have a single target "foo/bar" and we delete it, the "foo" folder will remain even though there's nothing left to be served in it.

Now, this change checks if the parent folder of the target we delete will be left empty, and if yes - deletes it too. In the example above where we delete a target "foo/bar" with location "./repository/targets/foo/bar" - deleting "bar" will delete the "foo" folder too (in case there are no more targets in it). So to your question - it is specific to that target.

The side effect that was caught by the tests is about when we have only 1 target "bar" in the repository with location "./repository/targets/bar" - deleting it will also delete its parent directory, which in this case is the "./repository/targets" folder. We need to decide if this is the expected behavior and if not, what is the correct way to handle that?

**Review by joshuagl — APPROVED** [review]
This change seems reasonable, it's hard to imagine all scenarios where this might cause breakage. I'm inclined to merge and handle error reports if they are filed.

**Review by trishankatdatadog — APPROVED** [review]

**Comment by trishankatdatadog** [comment]
> This change seems reasonable, it's hard to imagine all scenarios where this might cause breakage. I'm inclined to merge and handle error reports if they are filed.

Approve again as you see fit?

**Pull request merged**
